src/main/java/com/exactpro/th2/common/schema/grpc/router/impl/DefaultGrpcRouter.java

```diff
@@ -61,7 +65,21 @@
      * @throws ClassNotFoundException if matching the service class to protobuf stub has failed
      */
     public <T> T getService(@NotNull Class<T> cls) throws ClassNotFoundException {
-        return getProxyService(cls);
+        Objects.requireNonNull(cls, "Services class can not be null");
+
+        TH2Impl implementationAnnotation = cls.getAnnotation(TH2Impl.class);
+
+        if (implementationAnnotation == null) {
+            return getProxyService(cls);
+        }
+
+        Class<?> th2ImplClass = implementationAnnotation.value();
```

Comment:
I think we need to check if it is `null`

Reply:
Up. The class can be `null` (nobody will keep the users from setting it to null - we are not in Kotlin).